apps/dashboard/app/javascript/script_edit.js

```diff
@@ -148,38 +148,82 @@ function addInProgressField(event) {
   enableNewFieldButton();
 }
 
+function fixExcludeBasedOnSelect(selectElement) {
+  const excludeElementId = selectElement.dataset.excludeId;
+  const selectOptions = Array.from(selectElement.options);
+  const itemsToExclude = selectOptions.filter(opt => !opt.selected).map(opt => opt.text);
+  const excludeElement = document.getElementById(excludeElementId);
+  excludeElement.value = itemsToExclude.join(',');
+}
+
 function fixedFieldEnabled(checkbox, dataElement) {
   // Disable the element to avoid updates from the user
   dataElement.disabled = true;
   // As it is disabled, need to add a hidden field with the same name to send the fixed field value to the backend.
   const input = $('<input>').attr('type','hidden').attr('name', dataElement.name).attr('value', dataElement.value);
   $(checkbox).after(input);
+
+  if (dataElement.nodeName == 'SELECT') {
```

Comment:
I cannot find a better way to do this - but I'm calling it out as a `stringly typed language` code smell for me.

That said - I don't think there's anything to do though...

Reply:
I can add a flag into the fixed checkbox input in the template when it is used for a select field. And change the JS to check for that flag instead.

```js
if (dataElement.dataset.select) {
...
```

Reply:
I would have to check if this errors, but if it's safe then yes this is preferable. But again, it's got to be safe - so I'd have to check on that.

Reply:
I know we merged this as is, but while it's on my mind, I found a similar need in another feature. 

HTMLInputElement types have `type` API that I think we should use. I think this because a checkboxes `nodeName` and `tagName` both return `INPUT` where the `type` returns `checkbox`.  Similar thing with text inputs.

Just FYI.